### ISSUE TITLE
replaceAll on Elements does not work

### ISSUE THREAD

**Issue by EugenMayer** [jhy/jsoup#1514]
While this wont work:
```java
document.select("foo").replaceAll(element -> {
  return new Element("replacement");
});
```

this will work without any issues
```java
document.select("foo").forEach(element -> {
  element.replaceWith(new Element("replacement"));
});
```

If that makes any difference, in my concrete code i use 

```java
document.select(new Evaluator.Tag("pc:foo")).replaceAll(..);
```

But as far as i understand the implementation, that should not really matter

**Comment by EugenMayer** [comment]
Just happened to notice that `replaceAll` is not at all implemented but inherited from `ArrayList` on `Elements` - seems somewhat very surprising offering a custom collection implementation which offers methods that will actually not lead to expected result.

I understand that this issue is then somewhat a feature-request/grey area bug but maybe needs at least to be addressed by: 
 - hiding non supported operations like this one
 - implementing the method

**Comment by mct10** [comment]
Hi, I tried to implement this basically by overriding <code>replaceAll</code>. I think <code>node.replaceWith()</code> must be called, so as to change the element in the document. Simply calling <code>replaceAll</code> merely modifies the elements returned by <code>select</code>, leaving document itself unchanged. So, I call <code>replaceWith</code> in the new <code>replaceAll</code> and I think it works.

**Comment by jhy** [comment]
Thanks - I implemented this in #2017 and the similar methods (like set, clear, replaceAll etc) to also update the backing DOM, as well as the Elements list.
